server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -197,12 +207,23 @@ public long getLength() {
 
         public static UploadedSegmentMetadata fromString(String uploadedFilename) {
             String[] values = uploadedFilename.split(SEPARATOR);
-            return new UploadedSegmentMetadata(values[0], values[1], values[2], Long.parseLong(values[3]));
+            UploadedSegmentMetadata metadata = new UploadedSegmentMetadata(values[0], values[1], values[2], Long.parseLong(values[3]));
+            metadata.setWrittenBy(values[4]);
+
+            return metadata;
         }
 
         public String getOriginalFilename() {
             return originalFilename;
         }
+
+        public void setWrittenBy(String writtenBy) {
```

Comment:
We don't need a separate setter methods. We can provide the value in the constructor itself.

Reply:
The current structure makes it difficult to pass `writtenBy` during instantiation. While not ideal, we need to use a setter until we find a better approach. Related discussion: https://github.com/opensearch-project/OpenSearch/pull/8088#discussion_r1231740765

Reply:
Resolving this, please feel free to re-open if any update is needed.
Thanks!